bot/grenouille_http_server.py

```diff
@@ -0,0 +1,47 @@
+from http.server import BaseHTTPRequestHandler, HTTPServer
+import threading
+import logging
+import cgi
+
+
+
+class testHTTPServer_RequestHandler(BaseHTTPRequestHandler):
+
+    def do_POST(self):
+        form = cgi.FieldStorage(
+            fp=self.rfile,
+            headers=self.headers,
+            environ={"REQUEST_METHOD": "POST"}
+        )
+            
+        for item in form.list:
+            if (item.name == "say"):
+                bot.send_msg(item.value)
+                
+                self.send_response(200)
+                self.send_header("Content-type", "text/html")
+                self.end_headers()
+                return
+
+class GrenouilleHttpServer:
+    
+    def __init__(self, grenouille_bot):
+        self.grenouille_bot = grenouille_bot
+        
+    def start(self, grenouille_irc_bot):
+        global bot
```

Comment:
I would use a closure, kind of factory to create your RequestHandler which will have `bot` and you will have access to it